miden-lib/asm/miden/sat/internal/prologue.masm

```diff
@@ -679,78 +679,99 @@ end
 proc.process_consumed_notes_data
     # load the consumed notes data onto the advice stack
     exec.layout::get_nullifier_com adv.push_mapval dropw
+    # => [...]
 
     # read the number of consumed notes from the advice provider
     adv_push.1
+    # => [num_notes, ...]
 
     # store the number of consumed notes
     dup exec.layout::set_total_num_consumed_notes
+    # => [num_notes, ...]
 
-    # assert the number of consumed notes is within limits
-    dup exec.constants::get_max_num_consumed_notes lte assert
+    # assert the number of consumed notes is within limits; since max number of consumed notes is
```

Comment:
`since max number of consumed notes is smaller than 2^32` 

Isn't that what you check here? If so, shouldn't you say then:

`check that max number of consumed notes is smaller than 2^32`

Reply:
Because we expected the number of notes to be smaller than $2^{32}$ we can use a combination of `u32assert2` and `u32lte` instead of a general `lte` (which would have been less efficient).

I re-worded the statement to make this more clear.